galaxy_importer/ansible_test/builders/pulp_build.py

```diff
@@ -15,7 +15,271 @@
 # You should have received a copy of the Apache License
 # along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
 
+import json
+import logging
+import os
+import requests
+import subprocess
+import time
+import uuid
 
-def build_image_with_artifact():
+from galaxy_importer import config
+from galaxy_importer import constants
+from galaxy_importer import exceptions
+from shutil import copy, copyfile, copyfileobj
+
+default_logger = logging.getLogger(__name__)
+config_data = config.ConfigFile.load()
+cfg = config.Config(config_data=config_data)
+API_CHECK_RETRIES = 360
+API_CHECK_DELAY_SECONDS = 1
+
+
+class Build(object):
     """Use pulp-container to build ansible-test image with artifact inside."""
-    return ''
+    def __init__(self, pulp_artifact_file, api_url, content_url, metadata, logger):
+        self.api_url = api_url
+        self.content_url = content_url
+        self.basic_auth = ('admin', 'admin')
+        self.pulp_artifact_file = pulp_artifact_file
+        self.metadata = metadata
+        self.shared_dockerfile = os.path.join(
```

Comment:
can we point to the image on quay rather than the Dockerfile defined outside `galaxy_importer`? This will not be available when you pip install `galaxy-importer` - it's only available on cloning the repo